src/main/java/org/opensearch/knn/training/FloatTrainingDataConsumer.java

```diff
@@ -25,23 +34,39 @@
  */
 public class FloatTrainingDataConsumer extends TrainingDataConsumer {
 
+    private final QuantizationConfig quantizationConfig;
+
     /**
      * Constructor
      *
      * @param trainingDataAllocation NativeMemoryAllocation that contains information about native memory allocation.
      */
     public FloatTrainingDataConsumer(NativeMemoryAllocation.TrainingDataAllocation trainingDataAllocation) {
         super(trainingDataAllocation);
+        this.quantizationConfig = trainingDataAllocation.getQuantizationConfig();
     }
 
     @Override
     public void accept(List<?> floats) {
-        trainingDataAllocation.setMemoryAddress(
-            JNIService.transferVectors(
-                trainingDataAllocation.getMemoryAddress(),
-                floats.stream().map(v -> ArrayUtils.toPrimitive((Float[]) v)).toArray(float[][]::new)
-            )
-        );
+        if (floats.isEmpty() == false
+            && quantizationConfig != QuantizationConfig.EMPTY
+            && quantizationConfig.getQuantizationType() != null) {
```

Comment:
can quantizationConfig be null? Also you might get away with `quantizationConfig != QuantizationConfig.EMPTY` as quanttype will be null for it